changelog/pending/20250212--engine--add-bulkencrypt-for-snapshot-serialization.yaml

```
@@ -0,0 +1,4 @@
+changes:
+- type: feat
+  scope: engine
+  description: Add BulkEncrypt for snapshot serialization
```

```suggestion
  description: Add batch encryption for snapshot serialization to increase performance working with many secrets
```

We're calling this batch encryption now, and I think writing it this way is maybe nicer for a user facing changelog.